### PR TITLE
Open source automation modules

### PR DESCRIPTION
**Pull request by a-leonardi**
Various modifications to allow for open-source of Sekoia's Playbook modules

### CODE REVIEW

File: sekoia_automation/cli.py [hunk 1]
```diff
@@ -136,5 +137,13 @@ def openapi_to_module(
     ).run()
 
 
+@app.command(name="sync-library")
+def sync_library(
+    library: str = typer.Argument("../../modules", help="Path to the library"),
+    module: str = typer.Argument("", help="Module to deploy"),
+    check_image_on_registry: bool = typer.Option(False, help="Wether to check registry for existing image"),
+):
+    SyncLibrary(library, module, check_image_on_registry).execute()
```
Comment (Darkheir) [inline sekoia_automation/cli.py:146]: Keep naming of  other commands

```suggestion
    modules_path: Path = typer.Argument(..., help="Path to the playbook modules"),
    module: str = typer.Argument("", help="Module to deploy"),
    check_image_on_registry: bool = typer.Option(False, help="Wether to check registry for existing image"),
):
    SyncLibrary(modules_path, module, check_image_on_registry).execute()
```

File: pyproject.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ PyYAML = "^6.0"
 Jinja2 = "^3.0.3"
 black = "*"  # To format files in cli tools
 prometheus-client = "^0.16.0"
+requests-mock = "^1.10.0"
```
Comment (Darkheir) [inline pyproject.toml:47]: Is already in `tool.poetry.group.dev.dependencies`, I don't think it should be installed when we need only the SDK

File: sekoia_automation/scripts/sync_library.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+                return f"{prefix}{b64encode(f.read()).decode('utf-8')}"
+
+        return None
+
+    def check_image_on_registry(
+        self, docker_image: str, docker_image_version: str
+    ) -> bool:
+        registry_pat: str = os.getenv("CONTAINER_REGISTRY_PAT") or ""
+        # TODO: see with Raphael about this user env var
+        registry_user: str = os.getenv("CONTAINER_REGISTRY_USER") or ""
```
Comment (Darkheir) [inline sekoia_automation/scripts/sync_library.py:211]: I think it should be moved as an argument/option

File: sekoia_automation/scripts/sync_library.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+            prefix = "data:image/png;base64,"
+            path_to_use = pnglogo_path
+
+        if path_to_use:
+            with path_to_use.open("rb") as f:
+                return f"{prefix}{b64encode(f.read()).decode('utf-8')}"
+
+        return None
+
+    def check_image_on_registry(
```
Comment (Darkheir) [inline sekoia_automation/scripts/sync_library.py:206]: I'm not sure about this part (how to make it work with multiple registries).
For now we can try with the current implementation :)

File: sekoia_automation/scripts/sync_library.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+            p
+            for p in library_path.iterdir()
+            if p.is_dir() and p.joinpath("manifest.json").exists()
+        ]
+        print(f"Loading {len(modules)} modules from library in database\n")
+
+        for module in modules:
+            self.load_module(library_path / module.name)
+
+        print()
```
Comment (Darkheir) [inline sekoia_automation/scripts/sync_library.py:299]: Empty print, not sure why

**Review by Darkheir — CHANGES_REQUESTED** [review]
Few changes are needed :)

File: sekoia_automation/scripts/sync_library.py (hunk 5 above)
Comment (a-leonardi) [reply sekoia_automation/scripts/sync_library.py:299]: Just to add empty lines for aesthetic purposes I guess? They were already there in the source script

File: sekoia_automation/cli.py (hunk 1 above)
Comment (a-leonardi) [reply sekoia_automation/cli.py]: Are you sure we can take a Path from command line? typer will take charge of the conversion from the string input?

File: sekoia_automation/scripts/sync_library.py (hunk 4 above)
Comment (a-leonardi) [reply sekoia_automation/scripts/sync_library.py:206]: 👍

File: sekoia_automation/scripts/sync_library.py (hunk 5 above)
Comment (Darkheir) [reply sekoia_automation/scripts/sync_library.py:299]: Fair enough

**Review by Darkheir — APPROVED** [review]

**Pull request merged**